### DISCUSSION TITLE
How do I check the Next Run of a Job on the Scheduler?

### DISCUSSION THREAD

**Discussion by vinicius507** [go-co-op/gocron#246] [Q&A] [answered]
Greetings,

while trying to get the time of the Next Run of a job I keep getting a SIGSEGV. This is a somewhat simplified version of the code:

```go
sched := gocron.NewScheduler(time.Local)

func initTasks() {
    if constants.ENVIRONMENT == "Prod" {
         t1, _ := sched.Cron("* 18 ? * MON").Do(task1)
         sched.StartAsync()
         utils.Log.Debug(t1.NextRun())
    }
}
```

When it runs the logger with `t1.NextRun()`, it causes segmentation fault:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x69bdbd]
```

What is it that I'm doing wrong? Thanks in advance.

**Comment by JohnRoesler** [comment, marked as answer]
Hi @vinicius507 you have the right idea here. I am wondering if something inside your code (the logger, or the task itself) is throwing that seg fault. I made a simple example showing the next run with that cron schedule and it prints out just fine:

```golang
func main() {
	sched := gocron.NewScheduler(time.Local)

	job, err := sched.Cron("* 18 ? * MON").Do(func() {})
	if err != nil {
		log.Fatal(err)
	}
	sched.StartAsync()
	log.Println(job.NextRun())
}
```

Have you tried running with a debugger to see where it's hitting that issue? Or just putting in some fmt.Println statements to see how far it's getting before hitting that seg fualt?``

**Reply by vinicius507** [reply]
The jobs run just fine. I've done some extensive testing with them, and should they throw any errors, it should just log it and return.

It really happens when trying to call `job.NextRun()`. I'll investigate further with a debugger.
Thanks a lot, though :)

**Reply by JohnRoesler** [reply]
Yeah - definitely let me know what you find. It could be that the job was removed somehow and so NextRun() resulted in a nil job.

**Reply by vinicius507** [reply]
It was a problem with the structure method that I was using to initialize jobs (the job was placed inside a Task structure).

Before:
```go
func (t Task) Setup(sched *gocron.Scheduler) {
    // [...]
    t.job, err := sched.Cron("* 18 ? * MON").Do(t.handler)
    // [...]
}
```

After fixing:
```go
func (t *Task) Setup(sched *gocron.Scheduler) {
    // [...]
    t.job, err := sched.Cron("* 18 ? * MON").Do(t.handler)
    // [...]
}
```
